lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -61,13 +51,14 @@ def mock_resolve(self: Path, strict: bool = False):
         monkeypatch.setattr(Path, "resolve", mock_resolve)
         monkeypatch.setattr(Path, "stat", self.mock_stat)
         obj = UnpackTarballs(server_config, make_logger)
-        result = obj.unpack_tarballs(self.min_size, self.max_size)
-        for tar in self.tarlist:
-            assert (
-                f"Tarball link, '{tar}', does not resolve to a real location: No such file or directory: '{tar}'"
-                in caplog.text
-            )
-        assert result.total == 2
+        result = obj.unpack_tarballs(min_size, max_size)
+        if result.total > 0:
+            for tar in self.tarlist:
+                assert (
+                    f"Tarball link, '{tar}', does not resolve to a real location: No such file or directory: '{tar}'"
+                    in caplog.text
+                )
+        assert result.total == total
```

Comment:
In the last case.. the tarballs size range `0-586` the tarballs processing is not allowed for the given range. so, the caplog.text is empty.. Comparing the given assertion message with an empty `caplog.text` is throwing error.

Reply:
I see:  the `self.tarlist` is generated by the _test_ and contains all the possible tarballs; however, because of the size restrictions, none of them are selected.  As a result, `result.total` is zero but (of course!) `self.tarlist` is _not_ empty.

The problem is that the test is an all-or-nothing proposition -- it's not prepared for there to be some successes and some failures in the same run -- and so if _any_ files are selected, then they should _all_ show up in the `caplog`, and, if no files are selected, then the caplog will be empty.

🤔